public/app/plugins/datasource/influxdb/components/ConfigEditor.tsx

```diff
@@ -0,0 +1,203 @@
+
+const httpModes = [{ label: 'GET', value: 'GET' }, { label: 'POST', value: 'POST' }];
+
+export type Props = DataSourcePluginOptionsEditorProps<InfluxOptions>;
+
+type InfluxDataSourceSettings = DataSourceSettings<InfluxOptions, InfluxSecureJsonData>;
+
+export interface State {
+  config: InfluxDataSourceSettings;
+}
+
+export class ConfigEditor extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { options } = this.props;
+
+    this.state = {
+      config: ConfigEditor.defaults(options),
+    };
+  }
+
+  static getDerivedStateFromProps(props: Props, state: State) {
+    return {
+      ...state,
+      config: ConfigEditor.defaults(props.options),
+    };
+  }
+
+  static defaults = (options: InfluxDataSourceSettings) => {
+    options.secureJsonData = options.secureJsonData || {};
+    options.secureJsonData.password = options.secureJsonData.password || '';
+    return options;
+  };
+
+  onDatabaseChange = (database: string) => {
+    this.props.onOptionsChange({
+      ...this.state.config,
+      database,
+    });
+  };
+
+  onUserChange = (user: string) => {
+    this.props.onOptionsChange({
+      ...this.state.config,
+      user,
+    });
+  };
+
+  onPasswordChange = (password: string) => {
+    this.props.onOptionsChange({
+      ...this.state.config,
+      secureJsonData: {
+        ...this.state.config.secureJsonData,
+        password,
+      },
+    });
+  };
+
+  onTimeIntervalChange = (timeInterval: string) => {
+    this.props.onOptionsChange({
+      ...this.state.config,
+      jsonData: {
+        ...this.state.config.jsonData,
+        timeInterval,
+      },
+    });
+  };
+
+  onResetPassword = () => {
+    this.props.onOptionsChange({
+      ...this.state.config,
+      secureJsonFields: {
+        ...this.state.config.secureJsonFields,
+        password: false,
+      },
+    });
+  };
+
+  onHttpModeSelect = (httpMode: SelectableValue<string>) => {
+    this.props.onOptionsChange({
+      ...this.state.config,
+      jsonData: {
+        ...this.state.config.jsonData,
+        httpMode: httpMode.value,
+      },
+    });
+  };
+
+  render() {
+    const { config } = this.state;
+    return (
+      <>
+        <DataSourceHttpSettings
+          showAccessOptions={true}
+          dataSourceConfig={config}
+          defaultUrl="http://localhost:8086"
+          onChange={this.props.onOptionsChange}
+        />
+
+        <h3 className="page-heading">InfluxDB Details</h3>
+        <div className="gf-form-group">
+          <div className="gf-form-inline">
+            <div className="gf-form">
+              <FormLabel className="width-10">Database</FormLabel>
+              <div className="width-20">
+                <Input
+                  className="width-20"
+                  value={config.database || ''}
+                  onChange={(event: ChangeEvent<HTMLInputElement>) => this.onDatabaseChange(event.target.value)}
```

Comment:
This will create an anonymous function. This anonymous function will change every time this Component re renders so causing it to also re render. Maybe change this to:
```suggestion
                  onChange={this.onDatabaseChange}
```

I suggest converting all other anonymous functions in component as well.